src/cli/Hibp.cpp

```diff
@@ -0,0 +1,221 @@
+        }
+    };
+
+    uint qHash(const Sha1Hash& b)
+    {
+        uint result;
+        std::memcpy(&result, &b.m_data[0], sizeof(result));
+        return result;
+    }
+
+    bool parseHibpLine(QFile& file, Sha1Hash& sha1, size_t& numPwns, bool& atEnd, QString& error)
+    {
+        auto failedTo = [&](const QString& action) {
+            error = QObject::tr("failed to %1").arg(action);
+            return false;
+        };
+
+        auto failedToIo = [&](const QString& action) {
+            error = QObject::tr("failed to %1: %2").arg(action).arg(file.errorString());
+            return false;
+        };
+
+        atEnd = false;
+
+        char hexSha1[sizeof(Sha1Hash::m_data) * 2];
+        qint64 rc = file.read(&hexSha1[0], sizeof(hexSha1));
+        if (rc == -1) {
+            return failedToIo(QObject::tr("read SHA-1"));
+        } else if (rc == 0) {
+            atEnd = true;
+            return true;
+        } else if (rc != sizeof(hexSha1)) {
+            return failedTo(QObject::tr("read SHA-1 (short read)"));
+        }
+
+        if (!Tools::hexDecode(&sha1.m_data[0], sizeof(sha1.m_data), &hexSha1[0], sizeof(hexSha1))) {
+            return failedTo(QObject::tr("hex-decode SHA-1"));
+        }
+
+        char c;
+        if (!file.getChar(&c)) {
+            return failedToIo(QObject::tr("read ':' character"));
+        }
+        if (c != ':') {
+            return failedTo(QObject::tr("parse line (expected ':')"));
+        }
+
+        numPwns = 0;
+        while (true) {
+            if (!file.getChar(&c)) {
+                return failedToIo(QObject::tr("read pwnage count"));
+            }
+
+            if (c == '\n' || c == '\r') {
+                break;
+            }
+
+            if (!('0' <= c && c <= '9')) {
+                return failedTo(QObject::tr("parse pwnage count"));
+            }
+
+            numPwns *= 10;
+            numPwns += (c - '0');
+        }
+
+        while (1 == file.peek(&c, 1) && (c == '\n' || c == '\r')) {
+            file.getChar(&c);
+        }
+
+        return true;
+    }
+
+    QString pwnageReport(const Entry& entry, size_t numPwns)
+    {
+        QStringList ancestors;
+        for (auto g = entry.group(); g && g != g->database()->rootGroup(); g = g->parentGroup()) {
+            ancestors << g->name();
+        }
+
+        QString path;
+        for (int i = ancestors.size() - 1; i >= 0; --i) {
+            path += ancestors[i];
+            path += " > ";
+        }
+        path += entry.title();
+
+        return QObject::tr("Password for entry '%1' has been leaked %2 times!").arg(path).arg(numPwns);
+    }
+
+} // namespace
+
+Hibp::Hibp()
+{
+    name = "hibp";
+    description = QObject::tr("Check which passwords have been leaked.");
+}
+
+int Hibp::execute(const QStringList& arguments)
+{
+    TextStream outputTextStream(Utils::STDOUT, QIODevice::WriteOnly);
+    TextStream errorTextStream(Utils::STDERR, QIODevice::WriteOnly);
```

Comment:
Name them `out` and `err`. I actually don't know why the other modules have the long names again. I renamed them a while ago.